src/Metadata/Driver/TypedPropertiesDriver.php

```diff
@@ -33,15 +33,20 @@ class TypedPropertiesDriver implements DriverInterface
      * @var string[]
      */
     private $whiteList;
+    /**
+     * @var DocBlockTypeResolver
+     */
+    private $docBlockTypeResolver;
 
     /**
      * @param string[] $whiteList
      */
-    public function __construct(DriverInterface $delegate, ?ParserInterface $typeParser = null, array $whiteList = [])
+    public function __construct(DriverInterface $delegate, DocBlockTypeResolver $docBlockTypeResolver, ?ParserInterface $typeParser = null, array $whiteList = [])
```

Comment:
do we need to pass it as external dependency ? from what is see is a hard dependency for this class (and there are no interfaces for it..).
I would suggest to instantiate it inside.

Another (nicer?) approach would be to let it implement a `DriverInterface` and implement this kind of behavior for all the `array` types, not only for the one detected by php 7.4.

Reply:
I have changed declaration to keep the depedency hidden. It will be instantiate inside now.

I did implement it as separate Driver. During the implementation I found out, that it will be weird from the user perspective, why something like this can be serialized

    /**
    * @var Product[]
    */
    private array $products;

and something like this not:

    /**
    * @var Product
    */
    private $product;

This could lead to confusion, so both are correct now and we will be serialized correctly. 
Second thing is it will lower barrier, as PHP 7.4 is not needed anymore in order to use this feature.
The last and probably most valuable benefit. It will be much easier to move to `JMS Serializer` from the code, not using typed properties, as there will be no need for users to annotate it with `@Type` in each of the class properties. 
So install and use, if you have types defined in DocBlock :)